cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -410,4 +410,45 @@ const char* convert_fromUTF8_binary(int64 context, const char* bin_in, int32 len
   return ret;
 }
 
+FORCE_INLINE
+const char* replace_utf8_utf8_utf8(int64 context, const char* text, int32 text_len,
+                                   const char* from_str, int32 from_str_len,
+                                   const char* to_str, int32 to_str_len, int32* out_len) {
+  // if from_str is empty or its length exceeds that of original string,
+  // return the original string
+  if (from_str_len == 0 || from_str_len > text_len) {
+    *out_len = text_len;
+    return text;
+  }
+
+  char* out = reinterpret_cast<char*>(gdv_fn_context_arena_malloc(context, 8000));
```

Comment:
why 8000? should this not be at the maximum text len..

Reply:
you potentially might want to avoid the malloc until you actually find the match below, then do a memcpy for all elements before the match is found.